Delivered-To: david.abdurachmanov@gmail.com
Received: by 2002:a1c:7911:0:0:0:0:0 with SMTP id l17csp441700wme;
        Wed, 23 Oct 2019 00:30:29 -0700 (PDT)
X-Google-Smtp-Source: APXvYqz5208hun1CGlcY5LqvpWKkMSSy3RKQ7TUqGoWOvC3T++9OFyKaXm1XCHdzvKm/4+CZuq30
X-Received: by 2002:aa7:9e88:: with SMTP id p8mr9282700pfq.10.1571815829726;
        Wed, 23 Oct 2019 00:30:29 -0700 (PDT)
Return-Path: <opensbi-bounces+david.abdurachmanov=gmail.com@lists.infradead.org>
Received: from bombadil.infradead.org (bombadil.infradead.org. [2607:7c80:54:e::133])
        by mx.google.com with ESMTPS id z18si17129751pge.172.2019.10.23.00.30.29
        for <david.abdurachmanov@gmail.com>
        (version=TLS1_3 cipher=TLS_AES_256_GCM_SHA384 bits=256/256);
        Wed, 23 Oct 2019 00:30:29 -0700 (PDT)
Received-SPF: pass (google.com: best guess record for domain of opensbi-bounces+david.abdurachmanov=gmail.com@lists.infradead.org designates 2607:7c80:54:e::133 as permitted sender) client-ip=2607:7c80:54:e::133;
Authentication-Results: mx.google.com;
       dkim=pass header.i=@lists.infradead.org header.s=bombadil.20170209 header.b=Zwof1chA;
       dkim=neutral (body hash did not verify) header.i=@wdc.com header.s=dkim.wdc.com header.b=IlCmTOPG;
       dkim=neutral (body hash did not verify) header.i=@sharedspace.onmicrosoft.com header.s=selector2-sharedspace-onmicrosoft-com header.b=klgGKba1;
       arc=fail (body hash mismatch);
       spf=pass (google.com: best guess record for domain of opensbi-bounces+david.abdurachmanov=gmail.com@lists.infradead.org designates 2607:7c80:54:e::133 as permitted sender) smtp.mailfrom="opensbi-bounces+david.abdurachmanov=gmail.com@lists.infradead.org";
       dmarc=fail (p=NONE sp=NONE dis=NONE) header.from=wdc.com
DKIM-Signature: v=1; a=rsa-sha256; q=dns/txt; c=relaxed/relaxed;
	d=lists.infradead.org; s=bombadil.20170209; h=Sender:
	Content-Transfer-Encoding:Content-Type:Cc:List-Subscribe:List-Help:List-Post:
	List-Archive:List-Unsubscribe:List-Id:MIME-Version:Message-ID:Date:Subject:To
	:From:Reply-To:Content-ID:Content-Description:Resent-Date:Resent-From:
	Resent-Sender:Resent-To:Resent-Cc:Resent-Message-ID:In-Reply-To:References:
	List-Owner; bh=GxaBtT1rb2aXLyv+qlK2ndAohMQJiKDdW+JBSNtQkN0=; b=Zwof1chAq7inI3
	zS3alfTptTIAsxk2yUpDpUHIfB209iq7qYS4I/C28ipyAuhmOWvHySZho01FQIyJwTvwAgaMmoy4+
	3u3xztiO251oXCteGGXurqPK8Xhuq8o3/I3o5dqAibxs89EqWF6o82+ceeZhSrICx2WfR3NFHsFpg
	rQ7TrdCIRJb7+qyj2ubO6jJp0j2SF0GR6I754//YO3LMhMhAASpz5dJJHOYnvL2NzfE0WwD/KgQBJ
	rPxZQMI4MDLSNpXRsWPTulSazRYq5LED/72BcvRXbjzNWB0JcWBcL5zwHP0i5RRDAJfSFOXdz/KwP
	d707/nvL0WBduaRmCbig==;
Received: from localhost ([127.0.0.1] helo=bombadil.infradead.org)
	by bombadil.infradead.org with esmtp (Exim 4.92.3 #3 (Red Hat Linux))
	id 1iNB69-0000Qc-O1; Wed, 23 Oct 2019 07:30:25 +0000
Received: from esa2.hgst.iphmx.com ([68.232.143.124])
 by bombadil.infradead.org with esmtps (Exim 4.92.3 #3 (Red Hat Linux))
 id 1iNB66-0000Pi-Hi
 for opensbi@lists.infradead.org; Wed, 23 Oct 2019 07:30:24 +0000
DKIM-Signature: v=1; a=rsa-sha256; c=simple/simple;
 d=wdc.com; i=@wdc.com; q=dns/txt; s=dkim.wdc.com;
 t=1571815840; x=1603351840;
 h=from:to:cc:subject:date:message-id:
 content-transfer-encoding:mime-version;
 bh=Z84wpCibLmcrXtWo6xMnkj1vUfHtG408zYJMRGEQ61w=;
 b=IlCmTOPG/B9cSEfqi+GX/9wfUTIXV11pG5ufYGcZpRjJtHp3cP8ebNcs
 V6OO0ay4Sgjpzxjd/i26hDU1xNRurgOUElMbuV6zE/iRV9x738iBacqUg
 4xOZERYLexcDYqmhUVfvFrRxpLvU/r7yI2onXrM4hMs0jouy9pIDFYpxm
 WrFp0m1T80kjE21oD0pItfZVCtI0hwe/D+Gdvuh7WXhTXKWxyE8INgQCo
 574OqTIDzGhj+WMM2xKhCk9SKhBnMU+VX5SOOVr7G8bWAvIfOiXVMau/P
 fRJk2to1eB4JhXp+uSsXbUAATIwk2mT/0Dl2ZdXBYSOPQ4C5++zqL3PSP g==;
IronPort-SDR: MMEMJjjzMQV/lFgJu8e+O+A/4OKMpXR5zq1DrQfP0xSxVDKkb6JgD141EGQq2WQIXMwFys+cU4
 rIdFuR86Ny14x2FI1ogyGWowCPLa1xCQtHi3skUJX6/QGj6GoE71koPlPCYz9UQwVXJQoQoiPm
 QWX6szq5mBfcQ8r1pd7+0VUQRPPa5ReMmWY276zfROOeg6GV5w3nko4BJtCBA53u5rGVwE45ZI
 kgZoD2LEwg/jBXwQCCJP0UYj6eWbwc7bleMphqCW3NzPciRHreVSVrn4/xQtQF50NWxw29T6oN
 GRY=
X-IronPort-AV: E=Sophos;i="5.68,220,1569254400"; d="scan'208";a="222245320"
Received: from mail-by2nam05lp2050.outbound.protection.outlook.com (HELO
 NAM05-BY2-obe.outbound.protection.outlook.com) ([104.47.50.50])
 by ob1.hgst.iphmx.com with ESMTP; 23 Oct 2019 15:30:36 +0800
ARC-Seal: i=1; a=rsa-sha256; s=arcselector9901; d=microsoft.com; cv=none;
 b=Q5wabCES56AvRN4/VQs/0hQknUkbPv9pgfxhlPuJyj+aRmLPaPeNDucYbdNeIJkq8Yj5RAzo7lOndAeQ2VtTo4FFiqIvjlSDEL5J04qyWgcOTokvGAov7BF2PUe6dupEDVKT+eHNQO7Fr+rGYsY0P/8xfNiT0X6S0/7lJ8OJC3cTHskIegNyqrzIIFbjPAfnZ95Q+EaFAgFliKF9aQH0osYMSrcISpDqIBW+vsHXwjQA0M4SI5Hnbuc0m6hhG8N4SBqQ4sMaFweoz4ExCgyAx/Oh+7FnSZh8sMo35VHFjiqvOZMmNkE50F7jRgnVD07z6Ub5VyEJYtDN7rGJEbU3LQ==
ARC-Message-Signature: i=1; a=rsa-sha256; c=relaxed/relaxed; d=microsoft.com; 
 s=arcselector9901;
 h=From:Date:Subject:Message-ID:Content-Type:MIME-Version:X-MS-Exchange-SenderADCheck;
 bh=CUsb+g2HwXWBJfpnULZUTp/SvWSdgmHDWTqKBOsDiXg=;
 b=gUA0xhLhqqEw8cPq5fuXhGO1JThy2qIbn7/DcIpWvJ6skekqoMH/FkZ/zUq/0E9moNrn95j8RPUpuGKYdlIMw65uVUbHPLDU/290gNt45vxd3ogQ87Tk7UPP7R0D8TpU2BdoUnAr1vO2+BfHHCElLjaTvS2aqHfT1zbI7x+rytDlMf6OVAUSYneELi18W2P319LqA5EgdUmg1QHK1CZ6FcUZdbTVphghND6gJAh334SaVEMK+/By0JWSLC2ZPov+QQG3yC+3E/n2kUXmmq7qPAgisu3XE4RoWkJ+bCu759XJ2qb4TsY8cANTyyMRRh2wIpP+DYWHQs6FRYaoBH7I8g==
ARC-Authentication-Results: i=1; mx.microsoft.com 1; spf=pass
 smtp.mailfrom=wdc.com; dmarc=pass action=none header.from=wdc.com; dkim=pass
 header.d=wdc.com; arc=none
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
 d=sharedspace.onmicrosoft.com; s=selector2-sharedspace-onmicrosoft-com;
 h=From:Date:Subject:Message-ID:Content-Type:MIME-Version:X-MS-Exchange-SenderADCheck;
 bh=CUsb+g2HwXWBJfpnULZUTp/SvWSdgmHDWTqKBOsDiXg=;
 b=klgGKba1qMsJqGkSocGuccbXNsgm8bO/A7mOrTLaJhQN8k/YvVunYDAdj9jF7VABgw/Py4bzY5ZYEhK7lofykNlrRGu4ZYtEYelLXtHG1uRA3Q0QSGbJFLRi8mHrqBtfrhIyi1enQGiyVss7YgwY95M4wy/SxvpafcX1hpNcDiQ=
Received: from MN2PR04MB6061.namprd04.prod.outlook.com (20.178.246.15) by
 MN2PR04MB6000.namprd04.prod.outlook.com (20.178.246.90) with Microsoft SMTP
 Server (version=TLS1_2, cipher=TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384) id
 15.20.2367.20; Wed, 23 Oct 2019 07:30:18 +0000
Received: from MN2PR04MB6061.namprd04.prod.outlook.com
 ([fe80::1454:87a:13b0:d3a]) by MN2PR04MB6061.namprd04.prod.outlook.com
 ([fe80::1454:87a:13b0:d3a%7]) with mapi id 15.20.2387.019; Wed, 23 Oct 2019
 07:30:18 +0000
From: Anup Patel <Anup.Patel@wdc.com>
To: Atish Patra <Atish.Patra@wdc.com>, Alistair Francis
 <Alistair.Francis@wdc.com>
Subject: [PATCH] firmware: Fix compile error for FW_PAYLOAD with latest GCC
 binutils
Thread-Topic: [PATCH] firmware: Fix compile error for FW_PAYLOAD with latest
 GCC binutils
Thread-Index: AQHViXO4+JD6Bfi5lEyv6PJbDcYIog==
Date: Wed, 23 Oct 2019 07:30:18 +0000
Message-ID: <20191023072955.32016-1-anup.patel@wdc.com>
Accept-Language: en-US
Content-Language: en-US
X-MS-Has-Attach: 
X-MS-TNEF-Correlator: 
x-clientproxiedby: MA1PR01CA0091.INDPRD01.PROD.OUTLOOK.COM (2603:1096:a00::31)
 To MN2PR04MB6061.namprd04.prod.outlook.com
 (2603:10b6:208:d8::15)
authentication-results: spf=none (sender IP is )
 smtp.mailfrom=Anup.Patel@wdc.com; 
x-ms-exchange-messagesentrepresentingtype: 1
x-mailer: git-send-email 2.17.1
x-originating-ip: [129.253.179.161]
x-ms-publictraffictype: Email
x-ms-office365-filtering-ht: Tenant
x-ms-office365-filtering-correlation-id: f9cea0bf-ccf8-4b59-b385-08d7578adaab
x-ms-traffictypediagnostic: MN2PR04MB6000:
x-ms-exchange-purlcount: 1
x-ms-exchange-transport-forked: True
x-microsoft-antispam-prvs: <MN2PR04MB60004EBF4FE3FDE999C8E8E98D6B0@MN2PR04MB6000.namprd04.prod.outlook.com>
wdcipoutbound: EOP-TRUE
x-ms-oob-tlc-oobclassifiers: OLM:6430;
x-forefront-prvs: 019919A9E4
x-forefront-antispam-report: SFV:NSPM;
 SFS:(10019020)(4636009)(136003)(39860400002)(366004)(396003)(376002)(346002)(199004)(189003)(966005)(50226002)(1076003)(71200400001)(99286004)(256004)(81156014)(71190400001)(6636002)(8936002)(6486002)(14454004)(81166006)(102836004)(478600001)(6306002)(6506007)(6512007)(386003)(8676002)(486006)(26005)(44832011)(476003)(36756003)(3846002)(54906003)(110136005)(2906002)(5660300002)(86362001)(186003)(6116002)(6436002)(66946007)(25786009)(2616005)(66446008)(66476007)(52116002)(64756008)(66556008)(4326008)(316002)(7736002)(66066001)(305945005);
 DIR:OUT; SFP:1102; SCL:1; SRVR:MN2PR04MB6000;
 H:MN2PR04MB6061.namprd04.prod.outlook.com; FPR:; SPF:None; LANG:en;
 PTR:InfoNoRecords; A:1; MX:1; 
x-ms-exchange-senderadcheck: 1
x-microsoft-antispam: BCL:0;
x-microsoft-antispam-message-info: aPRt2hDZIgELP3bCuL3te740NaxAoQfWXXf8pNMdmh3sIIf2MVV76c5+6aD1GA1gdG3WmEWpmjL66A3sIbQFbhoT8qufZ9lN5xUef5QD4B4S+RRTaRfbD+MMWLif53qPdqLg2w2HsiOtRO7QFprBsUIBHsKjMNY+R3xZeE7U3g1lPE26N4Y11mn1snsqIWHbFakme89rS11T8x1XHxlMnM1Nv8qatvVWegXNmxSj6xnF9xpYgXqDL8sUskPf71OdWCV+W0N+GxeMAyTaEaJEYnVnp9v+bHca5wZQPu87ONISoxjKssSS2vw2VOf8jsSjdKKJpTyp7aTtHsYral37gWFZYtmgcxZNZ0TwNn8Gqf9BY5bI21SYhb2qhEIpiCAN4PCuMliGUTOfDVk0C2Ved1QR827LbzDsjGhugOEvNA82Z9fO5f3DxNskoJnprLGFJhFuVNFxTsOjnAJfn2e/9xvpYzuoCOtQsylWO+Awy7o=
MIME-Version: 1.0
X-OriginatorOrg: wdc.com
X-MS-Exchange-CrossTenant-Network-Message-Id: f9cea0bf-ccf8-4b59-b385-08d7578adaab
X-MS-Exchange-CrossTenant-originalarrivaltime: 23 Oct 2019 07:30:18.6980 (UTC)
X-MS-Exchange-CrossTenant-fromentityheader: Hosted
X-MS-Exchange-CrossTenant-id: b61c8803-16f3-4c35-9b17-6f65f441df86
X-MS-Exchange-CrossTenant-mailboxtype: HOSTED
X-MS-Exchange-CrossTenant-userprincipalname: YjEmmpMFgnua1YJRPqIfwCGMc2K5Qs9lxOaGE6e56FNrfPyryZ16lPezOaEsBOeHLrfXohHCMXYkdJZkj0eU+w==
X-MS-Exchange-Transport-CrossTenantHeadersStamped: MN2PR04MB6000
X-CRM114-Version: 20100106-BlameMichelson ( TRE 0.8.0 (BSD) ) MR-646709E3 
X-CRM114-CacheID: sfid-20191023_003022_678632_3B6D11FB 
X-CRM114-Status: GOOD (  10.95  )
X-Spam-Score: -2.5 (--)
X-Spam-Report: SpamAssassin version 3.4.2 on bombadil.infradead.org summary:
 Content analysis details:   (-2.5 points)
 pts rule name              description
 ---- ---------------------- --------------------------------------------------
 -2.3 RCVD_IN_DNSWL_MED      RBL: Sender listed at https://www.dnswl.org/,
 medium trust [68.232.143.124 listed in list.dnswl.org]
 0.0 SPF_HELO_NONE          SPF: HELO does not publish an SPF Record
 -0.0 SPF_PASS               SPF: sender matches SPF record
 -0.1 DKIM_VALID Message has at least one valid DKIM or DK signature
 -0.1 DKIM_VALID_EF          Message has a valid DKIM or DK signature from
 envelope-from domain
 -0.1 DKIM_VALID_AU          Message has a valid DKIM or DK signature from
 author's domain
 0.1 DKIM_SIGNED            Message has a DKIM or DK signature, not necessarily
 valid
X-BeenThere: opensbi@lists.infradead.org
X-Mailman-Version: 2.1.29
Precedence: list
List-Id: <opensbi.lists.infradead.org>
List-Unsubscribe: <http://lists.infradead.org/mailman/options/opensbi>,
 <mailto:opensbi-request@lists.infradead.org?subject=unsubscribe>
List-Archive: <http://lists.infradead.org/pipermail/opensbi/>
List-Post: <mailto:opensbi@lists.infradead.org>
List-Help: <mailto:opensbi-request@lists.infradead.org?subject=help>
List-Subscribe: <http://lists.infradead.org/mailman/listinfo/opensbi>,
 <mailto:opensbi-request@lists.infradead.org?subject=subscribe>
Cc: David Abdurachmanov <david.abdurachmanov@sifive.com>,
 Damien Le Moal <Damien.LeMoal@wdc.com>, Anup Patel <Anup.Patel@wdc.com>,
 "opensbi@lists.infradead.org" <opensbi@lists.infradead.org>
Content-Type: text/plain; charset="us-ascii"
Content-Transfer-Encoding: 7bit
Sender: "opensbi" <opensbi-bounces@lists.infradead.org>
Errors-To: opensbi-bounces+david.abdurachmanov=gmail.com@lists.infradead.org

We get following compile error for FW_PAYLOAD with latest GCC
binutils:
fw_payload.o(.text+0x1961): 15 bytes required for alignment to 16-byte
boundary, but only 14 present

Further investigating, it turn-out to be a known issue with RISC-V
GCC binutils.
(Refer, https://github.com/riscv/riscv-gnu-toolchain/issues/298)

As a work-around, we disable relaxation when including DTB and
PAYLOAD binary in fw_payload.S.

Reported-by: David Abdurachmanov <david.abdurachmanov@sifive.com>
Signed-off-by: Anup Patel <anup.patel@wdc.com>
---
 firmware/fw_payload.S | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/firmware/fw_payload.S b/firmware/fw_payload.S
index 4b4527a..e873f75 100644
--- a/firmware/fw_payload.S
+++ b/firmware/fw_payload.S
@@ -85,6 +85,14 @@ fw_options:
 	add	a0, zero, zero
 	ret
 
+	/*
+	 * We disable relaxation because use of ".align"
+	 * and ".balign" can potentially generate compile
+	 * errors with latest RISC-V GCC Binutils.
+	 */
+	.option push
+	.option norelax
+
 #ifdef FW_PAYLOAD_FDT_PATH
 	.align 4
 	.section .text, "ax", %progbits
@@ -103,3 +111,5 @@ payload_bin:
 #else
 	.incbin	FW_PAYLOAD_PATH
 #endif
+
+.option pop
-- 
2.17.1


_______________________________________________
opensbi mailing list
opensbi@lists.infradead.org
http://lists.infradead.org/mailman/listinfo/opensbi
